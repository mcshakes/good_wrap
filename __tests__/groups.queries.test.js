const resolvers = require("../resolvers");

describe("[Query.groupsByName]", () => {
    const mockContext = {
        dataSources: {
            goodreadsAPI: { getGroupsByName: jest.fn() }
        }
    };

    it("looks up groups by name", async () => {
        const getGroupsByName = mockContext.dataSources.goodreadsAPI.getGroupsByName;

        getGroupsByName.mockReturnValueOnce([
            {
                name: "goodreads-librarians-group",
                id: 220,
                users_count: 116716
            }
        ]);

        const res = await resolvers.Query.groupsByName(null, { name: "goodreads-librarians-group" }, mockContext);


        let responseObj = [
            {
                name: "goodreads-librarians-group",
                id: 220,
                users_count: 116716
            }
        ]

        expect(res).toEqual(responseObj);
        expect(res).toMatchSnapshot();
    })
})
