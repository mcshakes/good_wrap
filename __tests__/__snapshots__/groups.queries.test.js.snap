// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Query.groupsByName] looks up groups by name 1`] = `
Array [
  Object {
    "id": 220,
    "name": "goodreads-librarians-group",
    "users_count": 116716,
  },
]
`;
